composer/callbacks/checkpoint_saver.py

```diff
@@ -301,119 +301,117 @@ def __init__(
-                    artifact_name = format_name_with_dist_and_time(self.artifact_name, state.run_name,
-                                                                   state.timestamp).lstrip('/')
-                    # Always overwrite for symlinks since we use the same filename for latest
-                    with tempfile.TemporaryDirectory() as tmpdir:
-                        symlink_filename = os.path.join(tmpdir, 'latest.symlink')
-                        create_symlink_file(artifact_name, symlink_filename)
-                        logger.file_artifact(
-                            log_level=log_level,
-                            artifact_name=symlink_artifact_name,
-                            file_path=symlink_filename,
-                            overwrite=True,
-                        )
-
-        timestamp = state.timestamp
-
-        self.saved_checkpoints.append((timestamp, checkpoint_filepaths))
+        is_deepspeed = is_model_deepspeed(state.model)
+
+        # save the checkpoint to the filename
+        filename = self.filename.format(state, is_deepspeed)
+
+        saved = checkpoint.save_checkpoint(
+            state=state,
+            filename=filename,
+            weights_only=self.weights_only,
+        )
+
+        if not saved:  # not all ranks save
+            return
+
+        if self.latest_filename is not None:
+            symlink = self.latest_filename.format(state, is_deepspeed)
+            os.makedirs(os.path.dirname(symlink), exist_ok=True)
+            try:
+                os.remove(symlink)
+            except FileNotFoundError:
+                pass
+            os.symlink(os.path.relpath(filename, os.path.dirname(symlink)), symlink)
+
+        # if artifact name provided, upload the checkpoint
+        if self.artifact_name is not None:
+            artifact_name = self.artifact_name.format(state, is_deepspeed).lstrip('/')
+
+            logger.file_artifact(log_level=log_level,
+                                 artifact_name=artifact_name,
+                                 file_path=filename,
+                                 overwrite=self.overwrite)
+
+            if self.latest_artifact_name is not None:
+                symlink_name = self.latest_artifact_name.format(state, is_deepspeed).lstrip('/') + '.symlink'
+
+                # create and upload a symlink file
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    symlink_filename = os.path.join(tmpdir, 'latest.symlink')
+                    create_symlink_file(artifact_name, symlink_filename)
+                    logger.file_artifact(
+                        log_level=log_level,
+                        artifact_name=symlink_name,
+                        file_path=symlink_filename,
+                        overwrite=True,
+                    )
+
+        self.saved_checkpoints += [filename]
```

Comment:
nit:
```suggestion
        self.saved_checkpoints.append(filename)
```
modifies list in place instead of creating a new one.